### ISSUE TITLE
Plugin jar at update site is corrupt

### ISSUE THREAD

**Issue by sebthom** [CodesAway/BEXCodeCompare#130]
I tried to install the plugin from the update site https://codesaway.info/eclipse but it fails with:
```python
!ENTRY org.eclipse.equinox.p2.director 0 0 2022-04-04 21:23:44.835
!MESSAGE Overall install request is satisfiable
!SUBENTRY 1 org.eclipse.equinox.p2.director 0 0 2022-04-04 21:23:44.836
!MESSAGE Add request for BEX Code Compare 0.5.0.202010271910 (info.codesaway.bex.feature.group 0.5.0.202010271910) is satisfiable
Installation failed.
An error occurred during the org.eclipse.equinox.internal.p2.engine.phases.CheckTrust phase.
        session context was:(profile=epp.package.java, phase=org.eclipse.equinox.internal.p2.engine.phases.CheckTrust, operand=, action=).
        Error reading signed content.
        Caused by:
        java.util.zip.ZipException: invalid distance too far back

!ENTRY org.eclipse.equinox.p2.engine 4 4 2022-04-04 21:23:44.886
!MESSAGE An error occurred during the org.eclipse.equinox.internal.p2.engine.phases.CheckTrust phase.
!SUBENTRY 1 org.eclipse.equinox.p2.engine 4 0 2022-04-04 21:23:44.888
!MESSAGE session context was:(profile=epp.package.java, phase=org.eclipse.equinox.internal.p2.engine.phases.CheckTrust, operand=, action=).
!SUBENTRY 1 org.eclipse.equinox.p2.engine 4 0 2022-04-04 21:23:44.889
!MESSAGE Error reading signed content.
!STACK 0
java.util.zip.ZipException: invalid distance too far back
        at java.base/java.util.zip.InflaterInputStream.read(InflaterInputStream.java:165)
        at java.base/java.util.zip.ZipInputStream.read(ZipInputStream.java:197)
        at java.base/java.util.jar.JarInputStream.read(JarInputStream.java:207)
        at java.base/java.util.zip.ZipInputStream.closeEntry(ZipInputStream.java:143)
        at java.base/java.util.zip.ZipInputStream.getNextEntry(ZipInputStream.java:121)
        at java.base/java.util.jar.JarInputStream.getNextEntry(JarInputStream.java:142)
        at java.base/java.util.jar.JarInputStream.getNextJarEntry(JarInputStream.java:179)
        at org.eclipse.osgi.internal.signedcontent.SignedContentFromBundleFile.getSignedEntries(SignedContentFromBundleFile.java:224)
        at org.eclipse.osgi.internal.signedcontent.SignedContentFromBundleFile.<init>(SignedContentFromBundleFile.java:192)
        at org.eclipse.osgi.internal.signedcontent.SignedBundleHook.getSignedContent(SignedBundleHook.java:170)
        at org.eclipse.equinox.internal.p2.engine.phases.CertificateChecker.checkCertificates(CertificateChecker.java:94)
        at org.eclipse.equinox.internal.p2.engine.phases.CertificateChecker.start(CertificateChecker.java:73)
        at org.eclipse.equinox.internal.p2.engine.phases.CheckTrust.completePhase(CheckTrust.java:63)
        at org.eclipse.equinox.internal.p2.engine.Phase.postPerform(Phase.java:254)
        at org.eclipse.equinox.internal.p2.engine.Phase.perform(Phase.java:105)
        at org.eclipse.equinox.internal.p2.engine.PhaseSet.perform(PhaseSet.java:50)
        at org.eclipse.equinox.internal.p2.engine.Engine.perform(Engine.java:80)
        at org.eclipse.equinox.internal.p2.engine.Engine.perform(Engine.java:48)
        at org.eclipse.equinox.internal.provisional.p2.director.PlanExecutionHelper.executePlan(PlanExecutionHelper.java:46)
        at org.eclipse.equinox.internal.provisional.p2.director.PlanExecutionHelper.executePlan(PlanExecutionHelper.java:27)
        at org.eclipse.equinox.internal.p2.director.app.DirectorApplication.executePlan(DirectorApplication.java:772)
        at org.eclipse.equinox.internal.p2.director.app.DirectorApplication.planAndExecute(DirectorApplication.java:762)
        at org.eclipse.equinox.internal.p2.director.app.DirectorApplication.performProvisioningActions(DirectorApplication.java:743)
        at org.eclipse.equinox.internal.p2.director.app.DirectorApplication.run(DirectorApplication.java:1066)
        at org.eclipse.equinox.internal.p2.director.app.DirectorApplication.start(DirectorApplication.java:1204)
        at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:203)
        at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:136)
        at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
        at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:401)
        at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:255)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:659)
        at org.eclipse.equinox.launcher.Main.basicRun(Main.java:596)
        at org.eclipse.equinox.launcher.Main.run(Main.java:1467)
        at org.eclipse.equinox.launcher.Main.main(Main.java:1440)
```

Manually downloading https://codesaway.info/eclipse/plugins/info.codesaway.bex_0.5.0.202010271910.jar and testing the zip file resulted in:
```sh
$ unzip -t info.codesaway.bex_0.5.0.202010271910.jar
Archive:  info.codesaway.bex_0.5.0.202010271910.jar
  End-of-central-directory signature not found.  Either this file is not
  a zipfile, or it constitutes one disk of a multi-part archive.  In the
  latter case the central directory and zipfile comment will be found on
  the last disk(s) of this archive.
unzip:  cannot find zipfile directory in info.codesaway.bex_0.5.0.202010271910.jar,
        and cannot find info.codesaway.bex_0.5.0.202010271910.jar.zip, period.
```

**Comment by CodesAway** [comment]
Thanks for reporting the issue. I've been having issues with my development environment (for way too long). I have some time this week and plan to look into it.

I sadly don't know when it will be resolved. I'm hoping I'll get something figured out this week. I thank you for your understanding.

**Comment by CodesAway** [comment]
@sebthom I think I have a work around. It works for me when I tested on my work laptop and personal laptop, which previously wasn't able to download BEX due to the corrupt files. 

Will you please test the following update site. If all looks good, I'll replace the actual update site as well.
https://codesaway.info/eclipse-beta


My fix / workaround was deleting the prior files on the update site and generating new ones (without old versions). For some reason, it looks like Eclipse was corrupting the file or something (to be honest I have no clue what caused the issue). I thought starting from scratch would fix the issue, and it seems to have done so.

Please let me know if the beta update site works for you and I'll update the actual update site as well. Thanks again for reporting the issue.

**Comment by sebthom** [comment]
Looks good, thanks!!

**Comment by CodesAway** [comment]
Fixed update site and 0.6 release resolved issue. Copied contents of beta update site to stable update site, so both are consistent.
